library/src/androidTest/java/com/owncloud/android/lib/resources/shares/CreateShareRemoteOperationIT.kt

```diff
@@ -22,40 +24,60 @@ class CreateShareRemoteOperationIT : AbstractIT() {
     @Before
     fun before() {
         val result = GetStatusRemoteOperation(context).execute(client)
-        Assert.assertTrue(result.isSuccess)
+        assertTrue(result.isSuccess)
         val data = result.data as ArrayList<Any>
         val ownCloudVersion = data[0] as OwnCloudVersion
         Assume.assumeTrue(ownCloudVersion.isNewerOrEqual(NextcloudVersion.nextcloud_24))
     }
 
+    @Test
+    fun createShareWithNoteAndAttributes() {
+        val attributes =
+            ShareAttributesJsonHandler.toJson(
+                listOf(ShareAttributes.createDownloadAttributes(true))
+            )
+        val note = "Note with attributes"
+        val path = "/shareWithAttributes/"
+
+        createFolder(path)
+        val share = createShare(path, "admin", note, attributes)
+        assertEquals(note, share.note)
```

Comment:
I think that this test case should also check if the share attributes have been set correctly. Maybe something like this could work:

```suggestion
        val attributes = listOf(ShareAttributes.createDownloadAttributes(true))
        val note = "Note with attributes"
        val path = "/shareWithAttributes/"

        createFolder(path)
        val share = createShare(path, "admin", note, ShareAttributesJsonHandler.toJson(attributes))
        assertEquals(note, share.note)
        assertEquals(attributes, ShareAttributesJsonHandler.toList(share.attributes))
```